src/events/EventTileFactory.tsx

```diff
@@ -225,6 +225,10 @@ export function pickFactory(
             return noEventFactoryFactory(); // improper event type to render
         }
 
+        if (STATE_EVENT_TILE_TYPES[evType] === TextualEventFactory && !hasText(mxEvent, showHiddenEvents)) {
+            return noEventFactoryFactory();
+        }
```

Comment:
I am confused - do we just hide the event?

Reply:
`noEventFactoryFactory` is JSONEventFactory when show hidden events is enabled, see screenshot in the PR description

Reply:
NVM, /me was confused, sorry... Do I understand this correctly that you're hiding power-level events that have no diff?

Reply:
That's what already happened, this doesn't change that, events with no text don't render, this just applies that same logic to show hidden events and sticks the json view there

Reply:
Makes sense now - _resolved_